woocommerce/woocommerce
# [Accessibility] Add heading to coupon accordion in block-based Checkout page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #51796 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a product to the block-based Checkout page.
2. Go to the Checkout page.
3. Open the Developer tools and inspect the "Add a coupon" element on the Order Summary section on the right side.
4. Verify the `.wc-block-components-totals-coupon` element has the following attributes:
    - `role="heading"`
    - `aria-level="2"`
5. Test it with WP default themes and Storefront on mobile and Desktop.
6. Turn on the screen reader and focus on the "Add a coupon" element.
7. Verify the heading level 2 is announced.
![image](https://github.com/user-attachments/assets/ec6d345d-478f-4e04-833b-64272779af78)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
